### ISSUE TITLE
When triton loaded the python backend model preload libtensorflow_framework.so, a segment error

### ISSUE THREAD

**Issue by zhaohb** [triton-inference-server/server#5507]
**Triton Information**
What version of Triton are you using?  
22.08

Are you using the Triton container or did you build it yourself? 
Triton container

**To Reproduce**

My test model is an example provided by python backend: [https://github.com/triton-inference-server/python_backend/tree/main/examples/add_sub](url)
when I use that command:
```shell
LD_PRELOAD=/opt/tritonserver/backends/tensorflow2/libtensorflow_framework.so tritonserver --model-repository=add_repo/ --strict-model-config=false --log-verbose=0 --metrics-port=6000 --http-port=6609 --grpc-port=6601
```
the python backend model can not be load, and the demsg command outputs the following error：
```shell
[20646514.682790] triton_python_b[108867]: segfault at 0 ip 00007f975aa2e6e5 sp 00007ffdabc70d78 error 4 in libc-2.31.so[7f975a8c8000+178000]
[20646514.682797] Code: 00 00 0f 1f 00 31 c0 c5 f8 77 c3 66 2e 0f 1f 84 00 00 00 00 00 f3 0f 1e fa 89 f9 48 89 fa c5 f9 ef c0 83 e1 3f 83 f9 20 77 2b <c5> fd 74 0f c5 fd d7 c1 85 c0 0f 85 eb 00 00 00 48 83 c7 20 83 e1
```

but if I use
```shell
tritonserver --model-repository=add_repo/ --strict-model-config=false --log-verbose=0 --metrics-port=6000 --http-port=6609 --grpc-port=6601
```
model can be loaded normally

**Expected behavior**
It is hoped that the python backend model can be loaded normally with preload libtensorflow_framework.
Because we may customize some tf operators, which will depend on libtensorflow_framework.so, if this problem is not solved, the tf model and the python backend model cannot be loaded at same time

**Comment by Tabrizian** [comment]
I think the reason is that there will be symbol collisions with the TF backend. Python backend spawns the Python models in a separate process. As a workaround, do you think you can load the library in the `initialize` function of your Python model?

**Comment by zhaohb** [comment]
@Tabrizian Thank you for your response, but I don't think this is a good solution. We still need to solve the issue of symbol conflict between the Python backend and libtensorflow_framework.so. The initialize function should be used to initialize some information related to the Python backend model.
There is another scenario where both the Python backend model and TensorFlow model are loaded by Triton, But the TensorFlow model depends on libtensorflow_framework.so.

**Comment by Tabrizian** [comment]
> We still need to solve the issue of symbol conflict between the Python backend and libtensorflow_framework.so.

There isn't any symbol conflict between Python Backend and libtensorflow_framework.so. As mentioned, Python backend spawns the Python models in a separate process.

> There is another scenario where both the Python backend model and TensorFlow model are loaded by Triton, But the TensorFlow model depends on libtensorflow_framework.so.

There could be symbol conflicts between libtensorflow_framework.so and Tensorflow models but as mentioned above, Python backend loads models in a separate process and as a result it won't have symbol conflicts with libtensorflow_framework.so used by TF backend. We could probably allow custom ops support to Python backend similar to TRT backend (CC @dyastremsky).

**Comment by dyastremsky** [comment]
We support custom operations for TensorFlow models in the TF backend: https://github.com/triton-inference-server/server/blob/main/docs/user_guide/custom_operations.md#tensorflow

I don't think we have as standard practice to load the Triton backend via LD_PRELOAD when using Triton. That could result in unexpected behavior. Would it be possible to just load the TensorFlow library with the needed functions? It's also possible that the above is failing if newly spawned processes don't properly work with LD_PRELOAD, but Iman would be able to give more definitive guidane there.

@Tabrizian Currently, we support these in the respective backends using the APIs of those frameworks. Do we want to put in a feature request to support these in Python? It seems more challenging, as Python models could be anything. And they get loaded in their own process. We can talk more about this, but supporting this across any model is very different than supporting it for a model that uses a given framework (for which we can use that framework to load custom operators).

**Comment by zhaohb** [comment]
@dyastremsky Yes, using LD_PRELOAD to load custom ops is not a good approach as it can easily lead to symbol ,[https://github.com/triton-inference-server/server/blob/main/docs/user_guide/custom_operations.md#tensorflow](url)
this is a good approach，I will use that, thank you very much.
